common/src/main/java/com/sequenceiq/cloudbreak/common/service/LockService.java

```diff
@@ -0,0 +1,47 @@
+package com.sequenceiq.cloudbreak.common.service;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LockService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LockService.class);
+
+    @Inject
+    private JdbcTemplate jdbcTemplate;
+
+    public void lockAndRunIfLockWasSuccessful(Runnable runnable, Integer lockNumber) {
+        LOGGER.info("Try to get PostgreSQL advisory lock with lock number: {}", lockNumber);
+        boolean lockSuccess = lock(lockNumber, jdbcTemplate);
+        if (lockSuccess) {
+            LOGGER.info("PostgreSQL advisory lock was successful with lock number: {}", lockNumber);
+            runnable.run();
+            unlock(lockNumber, jdbcTemplate);
```

Comment:
What happens if the `runnable.run()` throws an exception and the unlock method won't be called? Does the lock stay there, and our services will go into a crash loop?
Can you please add some higher level test to make sure it will be unlocked even if the runnable fails?

Reply:
Thanks, good idea, I will put unlock in a finally block.